### ISSUE TITLE
Alternate text for Uploaded images

### ISSUE THREAD

**Issue by Tlazypanda** [publiclab/plots2#7991]
**Please describe the desired behavior.**
Currently we dont have any alternate text for images across the app. While for the static ones we can add some alt text but for the ones uploaded by the user, its not possible for us to apply the same. So what should be the plausible solution for it?

- One possible way is to add a field for the user to add alternate text for the image whenever they upload but this won't always ensure that the user does add the alternate text or if they do whether it will be meaningful text relevant to the picture. We can have a special message asking them to add since it will help accessibility of the app and help more sections of society to access it with ease
- The other solution is to use some image recognition service like aws/microsoft services to determine the image text and accordingly add it as alt text. This might slow down the app and might also require us to tag all our previous data or add our data to their servers.

**Comment by Tlazypanda** [comment]
@jywarren @cesswairimu @emilyashley @SidharthBansal @IshaGupta18 @VladimirMikulic Need your inputs everyone :tada: Thanks!! :v:

**Comment by VladimirMikulic** [comment]
Automation for me is always the way to go :)
Since this is OSS, I understand why other members might not agree with me. (AWS/Google/Microsoft holds the data).

@jywarren  is the right person to answer this question.

**Comment by Tlazypanda** [comment]
Hey @jywarren @cesswairimu @emilyashley @SidharthBansal @IshaGupta18  Looking forward to your inputs :tada:

**Comment by jywarren** [comment]
I hear you on automation sounding interesting! But I'm a bit hesitant on the automation front. It might be interesting to read up on and hear from folks who use accessibility software to know if automation (whether as a primary or fall-back method) has been a good solution for people -- and for people with various accessibility needs, for example. That would be a good "source of knowledge" for whether that would be a good approach. 

As to alt text and how to store it, it's also a challenge with Markdown. Commonmark, the closest thing to a Markdown standard, shows that alt text is provided for in Markdown: https://spec.commonmark.org/0.29/#example-568

Example: `![alt text](/image.png)`

The hard part may be to make it as easy (and encouraged) as possible for authors to add alt text. I think this is very worthwhile and more and more people are taking the time to add alt text to their images. This might mean some [coordination with the Editor team](https://github.com/publiclab/PublicLab.Editor/issues/490) over how the alt text should be added -- some brainstorming! Should images without alt text show a little icon overlay asking for alt content? Or something else?

One thing you could then do to have a hybrid approach is that where the editor asks for alt text, you could have a button that says "try to auto-generate a description with Google cloud blablabla" or something (😄 ) and the person would be able to review it to see if it sounds accurate. That way it's a supportive tool, not a default. 

Finally, remember that there's a comment editor and a post editor! People upload images in both. 

Thank you!!!! This is a really thoughtful issue and I appreciate your time and care on it!!!

**Comment by Tlazypanda** [comment]
Thanks @jywarren for such a detailed explanation :tada: Looping in @Shulammite-Aso @keshav234156 @NitinBhasneria @shreyaa-sharmaa What do you think? :sweat_smile:

**Comment by Shulammite-Aso** [comment]
If this is for the main images, maybe we could add an input field just beside the main image field, below the choose file button asking authors to include alternate text. And perhaps a help text too beside it that will hold something like this: 

> you could have a button that says "try to auto-generate a description with Google cloud blablabla" or something (smile ) and the person would be able to review it to see if it sounds accurate.

For images inside the text body, i don't actually know how exactly we could ask alt text from authors, but maybe if they use markdown, could replacing the 'image description' with their own description do the work?

**Comment by Tlazypanda** [comment]
Thanks @Shulammite-Aso for your inputs :tada: It does sound a bit challenging to figure this out :sweat_smile: Let's see how it fares

**Comment by Tlazypanda** [comment]
@jywarren Going through the different providers (microsoft,google etc) the services pricing structure requires us to pay :sweat_smile: It's either free for a certain trial period or paid according to the services or features of the api that you are going to require https://azure.microsoft.com/en-in/pricing/details/cognitive-services/computer-vision/

**Comment by cesswairimu** [comment]
Sure thing I will take a look
